rpc/errors.go

```diff
@@ -95,3 +100,15 @@ func (e *shutdownError) Code() int {
 func (e *shutdownError) Error() string {
 	return "server is shutting down"
 }
+
+// connectionClosedError is thrown when the connection is closed
+type connectionClosedError struct {
+}
+
+func (e *connectionClosedError) Code() int {
```

Comment:
I actually begin renaming this to `rpc.Error` since `rpc.RPCError` interface is a bit ambiguous. But in the process I decided to make a separate PR for it because there are more things that can be renamed and I didn't want to clutter this PR.

Not sure about renaming this method to `ErrorCode`. It would be something like `rpc.RPCError.ErrorCode`.